File: SEARCHING_AND_SORTING_17th_Feb_2023/linearSearch.cpp
#include <iostream>
using namespace std;

int linearSearch(int arr[], int n, int target) {
    for(int i = 0; i < n; i++) {
        if(arr[i] == target) {
            return i;
        }
    }
    return -1;
}

int main()
{
    int arr[] = {58, 69, 21, 33, 74, 10, 9, 79};
    int n = 8;
    int target;

    cout << "\nEnter the element you want to search: ";
    cin >> target;

    int ans = linearSearch(arr, n, target);
    
    cout << endl;
    cout << target << " is at " << ans << " index" << endl;
    return 0;
}

File: ARRAYS_10th_February_2023/LinearSearch.java
package ARRAYS_10th_February_2023;

/**
 * @author atulanand1811
 */
public class LinearSearch {
    public static void main(String[] args) {
        int arr[] = {15, 58, 22, 10, 6, 69, 36, 47};
        int key = 69;
        int flags = 0;
        for (int i = 0; i < arr.length; i++) {
            int search = arr[i];
            if (search == key){
                System.out.println(key+" is present at index "+i);
                flags = 1;
                break;
            }
        }
        if(flags == 0){
            System.out.println(key+" is not present in the array");
        }
    }
}
